apps/api/src/modules/job/application/offer/offer.service.ts

```diff
@@ -38,4 +38,48 @@ export class OfferService {
     )
     return { message: 'Job offer sent' }
   }
+
+  async acceptJobOffer(jobId: number, actorId: number) {
+    const job = await this.jobRepository.getBaseJobById(jobId)
+    if (!job) {
+      throw new NotFoundException('Job not found')
+    }
+    const application =
+      await this.applicationRepository.getApplicationbyActorJob(actorId, jobId)
+    if (!application) {
+      throw new BadRequestException('You have not applied to this job')
+    }
+    if (application.status === ApplicationStatus.OFFER_ACCEPTED) {
+      throw new BadRequestException('You have already accepted this job offer')
+    }
+    if (application.status !== ApplicationStatus.OFFER_SENT) {
+      throw new BadRequestException('You cannot accept this job offer')
+    }
```

Comment:
I think we don't need to check for specific status
If so you might done for other offer status
wait for @miello @PKhing to confirm